llvm/include/llvm/Frontend/HLSL/HLSLBinding.h

```diff
@@ -152,6 +171,12 @@ class BindingInfoBuilder {
         [&HasOverlap](auto, auto) { HasOverlap = true; });
   }
 
+  LLVM_ABI BoundRegs getBoundRegs() {
+    assert(std::is_sorted(Bindings.begin(), Bindings.end()) &&
+           "Bindings must be sorted");
```

Comment:
`std::is_sorted` already makes it clear what we're asserting. Might be better to explain why with something like "getBoundRegs should only be called after calculateBindingInfo"